src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -6076,96 +6097,164 @@ void ObjectMgr::LoadAccessRequirements()
 
     if (!_accessRequirementStore.empty())
     {
-        for (AccessRequirementContainer::iterator itr = _accessRequirementStore.begin(); itr != _accessRequirementStore.end(); ++itr)
-            delete itr->second;
+        for (DungeonProgressionRequirementsContainer::iterator itr = _accessRequirementStore.begin(); itr != _accessRequirementStore.end(); ++itr)
+        {
+            std::unordered_map<uint8, DungeonProgressionRequirements*> difficulties = itr->second;
+            for (auto difficultiesItr = difficulties.begin(); difficultiesItr != difficulties.end(); ++difficultiesItr)
+            {
+                for (auto questItr = difficultiesItr->second->quests.begin(); questItr != difficultiesItr->second->quests.end(); ++questItr)
+                {
+                    delete* questItr;
+                }
+
+                for (auto achievementItr = difficultiesItr->second->achievements.begin(); achievementItr != difficultiesItr->second->achievements.end(); ++achievementItr)
+                {
+                    delete* achievementItr;
+                }
+
+                for (auto itemsItr = difficultiesItr->second->items.begin(); itemsItr != difficultiesItr->second->items.end(); ++itemsItr)
+                {
+                    delete* itemsItr;
+                }
+
+                delete difficultiesItr->second;
+            }
+        }
 
         _accessRequirementStore.clear();                                  // need for reload case
     }
-
-    //                                               0      1           2          3          4     5      6             7             8                      9                  10
-    QueryResult result = WorldDatabase.Query("SELECT mapid, difficulty, level_min, level_max, item, item2, quest_done_A, quest_done_H, completed_achievement, quest_failed_text, item_level FROM access_requirement");
-    if (!result)
+    //                                                               0       1            2           3          4            5
+    QueryResult access_template_result = WorldDatabase.Query("SELECT id, map_id, difficulty, min_level, max_level, min_avg_item_level FROM dungeon_access_template");
+    if (!access_template_result)
     {
-        sLog->outString(">> Loaded 0 access requirement definitions. DB table `access_requirement` is empty.");
+        sLog->outString(">> Loaded 0 access requirement definitions. DB table `dungeon_access_template` is empty.");
         sLog->outString();
         return;
     }
 
+
     uint32 count = 0;
+    uint32 countProgressionRequirements = 0;
+
```

Comment:
Delete empty lines 6135 and 6138 as they are malicious blank spaces. (2 blank spaces in a row)